File: linkedlist/linkedlist.py
class Node:
    def __init__ (self, data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self, head=None):
        self.head = head

    def append(self, newNode):
        temp = self.head
        if self.head:
            while temp.next:
                temp = temp.next
            temp.next = newNode
        else:
            self.head = newNode

    def show(self):
        temp = self.head
        if self.head:
            while temp:
                print(temp.data,end=" ")
                temp = temp.next
        else: 
            print("There is no linked list")

e1 = Node(3)
e2 = Node(5)
e3 = Node(12)
e4 = Node(1)

ll = LinkedList(e1)
ll.append(e2)
ll.append(e3)
ll.append(e4)

ll.show()

File: linkedlist/linkedlist.cpp
#include<iostream>

// declaring the node structure
struct node {
    int data; 
    node* link;
}

// initialize a pointer to NULL
node* = A;
A = NULL;

// declaring a first node
node* temp = new node();
temp->data = 2;
temp->link = NULL;

// let A be pointer to the first node
A = temp;


// new node
temp = new node();
temp->data = 4;
temp->link = NULL;

// add new node to linked list
node* temp1 = A;
while (temp1->link != NULL) {
    temp1 = temp1->link;
}
temp1->link = temp;

/* Address of head node, A, is never modified */
